### PR TITLE
Add test infrastructure and marching cubes unit tests

### PR DESCRIPTION
**Pull request by schornakj**
- Add the `yak_ext` package to handle building and installing external dependencies like GTest.
- Add dependency on `ros-industrial/cmake_common_scripts` to take advantage of test setup macros.
- Add an initial collection of unit tests for the `yak_marching_cubes` library.
- Run tests during CI.

### CODE REVIEW

**Comment by schornakj** [comment]
The unit tests run in a separate script after the build process instead of during the `run_target_test` step, which is a little annoying but CI correctly fails if a unit test fails which produces the desired behavior.

**Comment by Levi-Armstrong** [comment]
> The unit tests run in a separate script after the build process instead of during the `run_target_test` step, which is a little annoying but CI correctly fails if a unit test fails which produces the desired behavior.

I agree, thankfully this is resolved in noetic because industrial ci is using colcon.

File: yak/package.xml [hunk 1]
```diff
@@ -9,11 +9,15 @@
   <license>MIT</license>
 
   <depend>libpcl-all-dev</depend>
+  <depend>cmake_common_scripts</depend>
```
Comment (Levi-Armstrong) [inline yak/package.xml]: This can just be a build_depend

File: yak_ext/package.xml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0"?>
+<package format="2">
+  <name>yak_ext</name>
+  <version>0.1.0</version>
+  <description>Provides external libraries for Yak.</description>
+  <maintainer email="joseph.schornak@swri.org">Joseph Schornak</maintainer>
+  <author email="joseph.schornak@swri.org">Joseph Schornak</author>
+  <license>Apache 2.0</license>
+  <depend>cmake_common_scripts</depend>
```
Comment (Levi-Armstrong) [inline yak_ext/package.xml]: Same here just a build depend.

**Review by Levi-Armstrong — APPROVED** [review]
Looks good, I would just change cmake_common_scripts to a build_depend.

**Comment by schornakj** [comment]
@Levi-Armstrong Thank you for your review!

**Pull request merged**